src/controllers/orders.ts

```diff
@@ -0,0 +1,57 @@
+import { NextFunction, Request, RequestHandler, Response } from "express";
+import { User } from "../models/User";
+import { Order } from "../models/Order";
+import { Cart } from "../models/Cart";
+import { GeneralError } from "../errors/general-error";
+import { postOrdersValidator } from "../validators/post-orders-validator";
+import { ValidationError } from "../errors/validation";
+import { ValidationErrorItem } from "sequelize";
+
+export const postOrders: RequestHandler = async (
+  request: Request,
+  response: Response,
+  next: NextFunction
+) => {
+  try {
+    postOrdersValidator(request.body);
+    const { transactionId, cartId, email } = request.body;
+    const user = request.user as User;
+    let order;
+    const cart = await Cart.findOne({ where: { id: cartId, status: 0 } });
+    if (!cart) return next(new GeneralError("Cart does not exist", 404));
+    cart.status = 1;
```

Comment:
Better to use ENUMs than writing the number yourself. Since it might be missed what each number means.

Define an enum in another file for types, example:
```
export enum CART_STATUS {
  ACTIVE = 1
  COMPLETED = 2
}
```

So you can use the enums like:
```
cart.status = CART_STATUS.ACTIVE;
```

and the cart status value will be `1`

Reply:
Yeah that's sounds like a good idea 💡
on it 👍